src/main/java/com/google/research/bleth/servlets/EnqueueSimulationServlet.java

```diff
@@ -0,0 +1,60 @@
+package com.google.research.bleth.servlets;
+
+import com.google.cloud.tasks.v2.AppEngineHttpRequest;
+import com.google.cloud.tasks.v2.CloudTasksClient;
+import com.google.cloud.tasks.v2.HttpMethod;
+import com.google.cloud.tasks.v2.QueueName;
+import com.google.cloud.tasks.v2.Task;
+import com.google.protobuf.ByteString;
+
+import java.io.IOException;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet used for enqueuing a task targeted at endpoint '/new-simulation',
+ * in order to create and run a new simulation.
+ */
+@WebServlet("/enqueue-simulation")
+public class EnqueueSimulationServlet extends HttpServlet {
+
+    static final String PROJECT_ID = "bleth-2020";
+    static final String LOCATION_ID = "europe-west1";
+    static final String QUEUE_ID = "simulations-queue";
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+        // If servlet runs on localhost, forward request without creating and enqueuing a task.
+        if (request.getServerName().equals("localhost") && request.getServerPort() == 8080) {
+            RequestDispatcher dispatcher = getServletContext()
+                    .getRequestDispatcher("/new-simulation");
+            dispatcher.forward(request, response);
+        } else {
```

Comment:
Prefer to just add a "return" into the "then" block, so the "else" part can just be unconditional.

This makes the code easier to read - a reader can know that the localhost case is completely done, and there's nothing later (outside the conditional) that still needs to happen. So they can now concentrate fully on the regular path.